bot/utils/redis_cache.py

```diff
@@ -54,8 +84,11 @@ def __get__(self, instance: RedisCache, owner: Any) -> RedisCache:
         if self.bot:
             return self
 
+        if self._namespace is None:
+            raise RuntimeError("RedisCache must be a class attribute.")
+
         if instance is None:
-            raise NotImplementedError("You must create an instance of RedisCache to use it.")
+            raise RuntimeError("You must create an instance of RedisCache to use it.")
```

Comment:
This exception message is incorrect, as `instance` does not mean the instance of `RedisCache` (that would be `self`).

What happens is that we need the cache to be accessed at least once through an instance of the **cog** because we grab the `Bot` instance from the cog. So, say that we have this cog:

```py
class MyCog:
    cache = RedisCache()
```

We need the `RedisCache` to be accessed as an attribute of a `MyCog` instance, like `self.cache` within one of the `MyCog` methods. That's the step that ensures that we can fetch the `bot` attribute (or whatever it is called) from the `MyCog` instance.

This `RuntimeError` only happens when something that will be incredibly rare happens: Someone tries to access the `cache` attribute on the *class* object directly (either `MyCog.cache` or the even more convoluted `self.__class_.cache` from within a cog method) **before** the first access through `self.cache` directly.

Long story, short: It's not about creating a `RedisCache` instance, but rather that the `RedisCache` instance **has to be** accessed through an attribute of the *cog instance* (`self.cache`) the *very first time* it's accessed so we can grab the `Bot` reference.

```suggestion
            raise RuntimeError(
                "You must access the RedisCache instance through the cog instance "
                "before accessing it using the cog's class object."
            )
```